cmd/genji/shell/shell.go

```diff
@@ -300,6 +355,23 @@ func (sh *Shell) getDB() (*genji.DB, error) {
 	return sh.db, nil
 }
 
+// buildJsonString converts string json array to string json document
+// [{"a":1}, {"b":2}] => {"":[{"a":1}, {"b":2}]}.
+func buildJsonString(data string) string {
```

Comment:
We should not use regexp to transform json.
Use [json.RawMessage](https://golang.org/pkg/encoding/json/#RawMessage) instead:

```go
var array []json.RawMessage
// then decode to that variable
```

Reply:
Going to see.

Reply:
Even if I use `[]json.RawMessage` I get an error when I insert the datas that contain `\r`. 
`error: found \r, expected } at line 65, char 85`

The only way, I know is to use regexp.
So the `json.RawMessage` keep the best way decode array json.
thanks.

Reply:
Can you provide an example here?

Reply:
It happens when I run.
`curl https://api.github.com/repos/genjidb/genji/issues | genji insert -f my.db -t foo`

Reply:
Can you give me more details please? Like what piece of code is reading the data. Ideally, write a simplified example in a comment here with juste the relevant code and input.

Reply:
```
# reads input
in, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		return fmt.Errorf("unable to read piped input: %w", err)
	}

return executeInsertCommand(db, table, string(in))
```

```
# here is executed the query and got error.
func executeInsertCommand(db *genji.DB, table, data string) error {
	var b strings.Builder
	/*
         # filing b with INSERT INTO table VALUES 
       */
var jRaw []json.RawMessage
	if err := json.Unmarshal([]byte(data), &jRaw); err != nil {
		return err
	}

	for i := range jRaw {
		b.WriteString(string(jRaw[i]))
		err := db.Exec(b.String())
		if err != nil {
                     # getting error from db.Exec()
			return err
		}
		// Reset the buffer and fill it with new insert.
		b.Reset()
		b.WriteString(insert)
	}

  return nil
}
```

Reply:
You should not try to build a string for a query, use query parameters instead. Maybe json.RawMessage doesn't work here, try using a map:

```go
var m map[string]interface{}
json.Unmarshal(&m)
db.Exec("INSERT INTO foo VALUES ?", document.NewFromMap(&m))
```